lib/internal/fs/glob.js

```diff
@@ -68,6 +85,21 @@ class Cache {
     this.#readdirCache.set(path, val);
     return val;
   }
+
+  async readdir(path) {
+    const cached = this.#readdirCache.get(path);
+    if (cached) {
+      return cached;
+    }
+    let val;
+    try {
+      val = await readdir(path, { __proto__: null, withFileTypes: true });
+    } catch {
+      val = [];
+    }
+    this.#readdirCache.set(path, val);
```

Comment:
If the goal is to streamline the process of finding test files, shouldn't we be using `opendir` rather than `readdir`?

Reply:
thanks, I need to check how I can insert it here, I did not change the logic...

Reply:
+1 for `opendir`